File: 8-kyu/is-it-a-palindrome-/c++/solution.cpp
#include <algorithm>
#include <string>

std::string makeLower(const std::string &str)
{
    auto str_cpy = str;
    for (char &c : str_cpy) {
        if (std::isalpha(c)) c = std::isupper(c) ? std::tolower(c) : c;
    }
    return str_cpy;
}

bool isPalindrom (const std::string& str)
{
    auto lower_str = makeLower(str);
    return std::equal(lower_str.begin(), lower_str.end(), lower_str.rbegin());
}

File: 8-kyu/is-it-a-palindrome-/python/solution.py
#include <string.h>

int isPalindrom(const char *s)
{
    if (strlen(s) == 1) return 1;
    size_t j = strlen(s) - 1, ret = 1;
    const size_t length = j;
    for (size_t i = 0; i < length; i++, j--) {
        if (tolower(s[i]) != tolower(s[j])) ret = 0;
    }
    return ret;
}